### PR TITLE
Genome size non-hacky :)

### PR DESCRIPTION
**Pull request by Maarten-vd-Sande**
**What problem is the PR solving / What's new?**
Before hacky genome size. Now reusable, and sensible approach :)

**Checklist**
- [x] I made a PR to develop (not master)
- [ ] If applicable: I updated the docs
- [x] I updated the CHANGELOG
- [x] These changes are covered by the tests

### CODE REVIEW

File: seq2science/rules/get_genome.smk [hunk 1]
```diff
@@ -158,3 +158,28 @@ rule gene_id2name:
...
+    Get the effective genome size for a kmer length. Some tools (e.g. macs2) require
+    an estimation of the effective genome size to better estimate how (un)likely it
+    is to have a certain number of reads on a position. The actual genome size is
+    not the best indication in these cases, since reads in repetitive regions
+    (for a certain kmer length) can not possible align on some locations.
+    """
+    input:
+        expand("{genome_dir}/{{assembly}}/{{assembly}}.fa", **config),
+    output:
+        expand("{genome_dir}/{{assembly}}/{{assembly}}.kmer_{{kmer_size}}.genome_size", **config)
```
Comment (siebrenf) [inline seq2science/rules/get_genome.smk:174]: how many files should I expect? If its more than 5, maybe make a subdirectory?

File: seq2science/rules/qc.smk [hunk 2]
```diff
@@ -116,11 +116,11 @@ if config["trimmer"] == "trimgalore":
             """
 
 
-elif config["trimmer"] == "fastp":
+elif config["trimmer"] == "fastp" or "kmer_estimation" in config:
```
Comment (siebrenf) [inline seq2science/rules/qc.smk:119]: what's with this `or`? We only support these 2 trimmer right?

File: seq2science/rules/qc.smk [hunk 3]
```diff
@@ -116,11 +116,11 @@ if config["trimmer"] == "trimgalore":
...
     ruleorder: fastp_qc_PE> fastp_qc_SE > fastp_PE > fastp_SE
 
 
-    rule fastp_qc_SE:
+    checkpoint fastp_qc_SE:
```
Comment (siebrenf) [inline seq2science/rules/qc.smk:123]: I don't think this one needs to be a checkpoint. It isn't referenced anywhere at least.

File: seq2science/rules/configuration_workflows.smk [hunk 4]
```diff
@@ -139,3 +140,47 @@ if config.get("contrasts"):
             f"\nCould not parse DESeq2 contrast '{contrast}'.\n"
             "A DESeq2 design contrast must be in the form '(batch+)column_target_reference'. See the docs for examples.\n")
         _,_,_,_ = parse_contrast(contrast, samples, check=True)
+
+
+def get_read_length(sample):
```
Comment (siebrenf) [inline seq2science/rules/configuration_workflows.smk:145]: Do the inputs need to be checkpoints? Would it not be faster to add these qc files/`rules.get_effective_genome_size.output` as input for macs2? My concern is speed here. If that is not applicable, then nvm :)

**Review by siebrenf — COMMENTED** [review]
In general, speeding up macs is nice! Not sure how fast/easy checkpoints are now though. Some of these don't look like they need to be checkpoints anyway!

File: seq2science/rules/configuration_workflows.smk (hunk 4 above)
Comment (Maarten-vd-Sande) [reply seq2science/rules/configuration_workflows.smk:145]: The read length is also used by e.g. the chromap aligner, which I'll implement now that we finally have working checkpoints. I don't think checkpoints have a big performance hindrance.

Also if we don't use checkpoints on estimation of the read lengths, we would have to estimate the genome size for each sample separately, which is the whole reason i did this :)

File: seq2science/rules/qc.smk (hunk 2 above)
Comment (Maarten-vd-Sande) [reply seq2science/rules/qc.smk:119]: Oops :see_no_evil:

File: seq2science/rules/qc.smk (hunk 3 above)
Comment (Maarten-vd-Sande) [reply seq2science/rules/qc.smk:123]: :+1:

**Review by siebrenf — APPROVED** [review]

**Pull request merged**